pkg/model/iam/pod_roles.go

```diff
@@ -19,10 +19,40 @@ package iam
 import (
 	"fmt"
 
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/kops/pkg/apis/kops"
 	"k8s.io/kops/pkg/featureflag"
+	"k8s.io/kops/pkg/wellknownusers"
 )
 
+// PodRole holds a pod-specific IAM policy
+type PodRole string
+
+const (
+	// PodRoleEmpty indicates we are not using a PodRole
+	PodRoleEmpty PodRole = ""
+	// PodRoleDNSController is the role used by the dns-controller pods
+	PodRoleDNSController PodRole = "dns-controller"
+)
+
+// PodOrNodeRole is used to specify generation of permissions for either a pod or a node
+type PodOrNodeRole struct {
```

Comment:
ServiceAccountOrNodeRole?

Reply:
...or is just `Role` possible?

Reply:
It's now `iam.Subject` now that it's an interface - I think that's the correct term!